packages/edit-site/src/components/save-button/index.js

```diff
@@ -21,24 +24,30 @@ export default function SaveButton( {
 	icon,
 	__next40pxDefaultSize = false,
 } ) {
-	const { isDirty, isSaving, isSaveViewOpen } = useSelect( ( select ) => {
-		const {
-			__experimentalGetDirtyEntityRecords,
-			isSavingEntityRecord,
-			isResolving,
-		} = select( coreStore );
-		const dirtyEntityRecords = __experimentalGetDirtyEntityRecords();
-		const { isSaveViewOpened } = select( editSiteStore );
-		const isActivatingTheme = isResolving( 'activateTheme' );
-		return {
-			isDirty: dirtyEntityRecords.length > 0,
-			isSaving:
-				dirtyEntityRecords.some( ( record ) =>
-					isSavingEntityRecord( record.kind, record.name, record.key )
-				) || isActivatingTheme,
-			isSaveViewOpen: isSaveViewOpened(),
-		};
-	}, [] );
+	const { isDirty, isSaving, isSaveViewOpen, previewingThemeName } =
+		useSelect( ( select ) => {
+			const {
+				__experimentalGetDirtyEntityRecords,
+				isSavingEntityRecord,
+				isResolving,
+			} = select( coreStore );
+			const dirtyEntityRecords = __experimentalGetDirtyEntityRecords();
+			const { isSaveViewOpened } = select( editSiteStore );
+			const isActivatingTheme = isResolving( 'activateTheme' );
+			const previewingTheme = select( coreStore ).getTheme(
+				currentlyPreviewingTheme()
+			);
+
+			return {
+				isDirty: dirtyEntityRecords.length > 0,
+				isSaving:
+					dirtyEntityRecords.some( ( record ) =>
+						isSavingEntityRecord( record.kind, record.name, record.key )
```

Comment:
There seems to be a prettier error on this line. 
https://github.com/WordPress/gutenberg/actions/runs/6730275641/job/18292658610?pr=55752#step:5:592

```
 Replace `·record.kind,·record.name,·record.key·` with `⏎↹↹↹↹↹↹↹record.kind,⏎↹↹↹↹↹↹↹record.name,⏎↹↹↹↹↹↹↹record.key⏎↹↹↹↹↹↹`  prettier/prettier
```

I don't think this has anything to do with this PR change, but it would be good to address it before merging! @arthur791004

Reply:
Updated!